src/components/LegacyMediaAnalytics/createMediaAnalyticsTracker.js

```diff
@@ -0,0 +1,277 @@
+      .then(sessionID => {
+        const xdm = {
+          eventType: getEventType({ eventType }),
+          mediaCollection: {
+            sessionID,
+            playhead: trackerState.lastPlayhead
+          }
+        };
+        deepAssign(xdm.mediaCollection, mediaDetails);
+        const event = mediaEventManager.createMediaEvent({ options: { xdm } });
+        return mediaEventManager.trackMediaEvent({ event }).then(() => {
+          updateLastTimeEventTriggered();
+        });
+      })
+      .catch(error => {
+        logger.info(error);
+      });
+  };
+
+  const heartbeatEngine = createHeartbeat({
+    mainPingInterval,
+    deferSession,
+    playerState: trackerState,
+    trackEvent
+  });
+
+  return {
+    trackSessionStart: (mediaObject, contextData = {}) => {
+      if (isEmptyObject(mediaObject)) {
+        deferSession.reject("invalid media object");
+        return {};
+      }
+
+      const mediaCollection = {
+        playhead: 0
+      };
+      const {
+        sessionDetails,
+        customMetadata
+      } = convertSessionDetailsMetadataToXDM(contextData);
+
+      deepAssign(
+        mediaCollection,
+        mediaObject,
+        { sessionDetails },
+        { customMetadata }
+      );
+      const event = mediaEventManager.createMediaSession({
+        xdm: {
+          mediaCollection
+        }
+      });
+
+      return mediaEventManager
+        .trackMediaSession({
+          event
+        })
+        .then(result => {
+          updateLastTimeEventTriggered();
+          deferSession.resolve(result.sessionId);
+
+          ticker = setInterval(() => {
+            heartbeatEngine();
+          }, 1000);
+        });
+    },
+    trackPlay: () => {
+      return trackEvent({ eventType: Event.Play });
+    },
+    trackPause: () => {
+      return trackEvent({ eventType: Event.Pause });
+    },
+    trackSessionEnd: () => {
+      clearInterval(ticker);
+      return trackEvent({ eventType: Event.SessionEnd }).then(() => {
+        deferSession = null;
+      });
+    },
+    trackComplete: () => {
+      clearInterval(ticker);
+      return trackEvent({ eventType: Event.SessionComplete });
+    },
+    trackError: errorId => {
+      logger.info(`trackError(${errorId})`);
+
+      const errorDetails = {
+        name: errorId,
+        source: "player"
+      };
+      const mediaDetails = {
+        errorDetails
+      };
+      return trackEvent({ eventType: Event.SessionEnd, mediaDetails });
+    },
+    trackEvent: (eventType, info, context) => {
+      if (isEmptyObject(info)) {
+        logger.info("invalid media object");
+        return {};
+      }
+      const mediaDetails = createMediaDetailsObject({
+        eventType,
+        info,
+        context
+      });
+
+      return trackEvent({
+        eventType,
+        mediaDetails
+      });
+    },
+    updatePlayhead: time => {
+      if (isNumber(time)) {
+        trackerState.lastPlayhead = parseInt(time, 10);
```

Comment:
Media Edge API has validation for `playhead`, it should be of type `Integer`. In my opinion, we should do the transformation in the library rather than throwing a validation exception.